parsl/dataflow/dflow.py

```diff
@@ -197,7 +197,10 @@ def handle_update(self, task_id, future, memo_cbk=False):
         # Submit _*_stage_out tasks for output data futures that correspond with remote files
         if (self.tasks[task_id]['app_fu'] and
                 self.tasks[task_id]['status'] == States.done and
-                self.tasks[task_id]['executor'] != 'data_manager'):
+                self.tasks[task_id]['executor'] != 'data_manager' and
+                self.tasks[task_id]['func_name'] != '_file_stage_in' and
```

Comment:
We should add a task type "stage_in" , "stage_out" tag rather than check func_names